src/memex/models/document.py

```diff
@@ -393,10 +395,14 @@ def merge_documents(session, documents, updated=None):
             m.document = master
             m.updated = updated
 
-        session.delete(doc)
-
     try:
         session.flush()
+        session.query(Annotation) \
+            .filter(Annotation.document_id.in_(duplicate_ids)) \
```

Comment:
Or rather, I will add the FIXME comment to use the relationship for filtering, but ... (see above for the rest)